src/cli/self_update/windows.rs

```diff
@@ -402,26 +411,33 @@ pub fn delete_rustup_and_cargo_home() -> Result<()> {
 
 #[cfg(test)]
 mod tests {
+    use std::ffi::OsString;
+    use std::os::windows::ffi::OsStrExt;
+
     use winreg::enums::{RegType, HKEY_CURRENT_USER, KEY_READ, KEY_WRITE};
     use winreg::{RegKey, RegValue};
 
     use crate::currentprocess;
     use crate::test::with_saved_path;
     use crate::utils::utils;
 
+    fn wide(str: &str) -> Vec<u16> {
+        OsString::from(str).encode_wide().collect()
```

Comment:
I think it can be simpler no?

`OsString::from_wide(&str[..]);`

I wouldn't even have the function